mmdet/integration/nncf/runners.py

```diff
@@ -0,0 +1,76 @@
+import warnings
+import time
+
+import mmcv
+from mmcv.runner import EpochBasedRunner
+from mmcv.runner import RUNNERS
+from mmcv.runner.utils import get_host_info
+from mmdet.core.evaluation.eval_hooks import EvalHook
+
+from .utils import check_nncf_is_enabled
+
+
+@RUNNERS.register_module()
+class AccuracyAwareRunner(EpochBasedRunner):
+    """
+    An mmdet training runner to be used with NNCF-based accuracy-aware training.
+    Inherited from the standard EpochBasedRunner with the overridden "run" method.
+    This runner does not use the "workflow" and "max_epochs" parameters that are
+    used by the EpochBasedRunner since the training is controlled by NNCF's
+    AdaptiveCompressionTrainingLoop that does the scheduling of the compression-aware
+    training loop using the parameters specified in the "accuracy_aware_training".
+    """
+    def __init__(self, *args, target_metric_name='bbox_mAP', **kwargs):
+        super().__init__(*args, **kwargs)
+        if not any([isinstance(hook, EvalHook) for hook in self._hooks]):
```

Comment:
eval hooks in this pipeline are registered through `runner.register_hook(eval_hook(val_dataloader, **eval_cfg))`, which means they are not available on initialization.

Reply:
Removed